lib/rules/no-invalid-html-attribute.js

```diff
@@ -291,12 +289,11 @@ function checkLiteralValueNode(context, attributeName, node, parentNode, parentN
     const allowedTags = VALID_VALUES.get(attributeName).get(singlePart.value);
     const reportingValue = singlePart.reportingValue;
 
-    const suggest = [
-      Object.assign(
-        getMessageData('suggestRemoveInvalid', messages.suggestRemoveInvalid),
-        { fix(fixer) { return fixer.removeRange(singlePart.range); } }
-      ),
-    ];
+    const suggest = [{
+      messageId: 'suggestRemoveInvalid',
+      data: { reportingValue },
+      fix(fixer) { return fixer.removeRange(singlePart.range); },
+    }];
```

Comment:
Similar to https://github.com/jsx-eslint/eslint-plugin-react/pull/3759#discussion_r1606710443:

![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/44349756/d4c3ad73-ca9c-4e76-941d-28fe68fc728f)

I fixed all suggestions in this rule, please check if the passed data makes sense.